### ISSUE TITLE
boolean string parsing behaviour in scala-native

### ISSUE THREAD

**Issue by fsaab** [scala-native/scala-native#1150]
what could be the reason for a normal parsing code to give on the scale-native side(after compiling) the below exception:

 java.lang.IllegalArgumentException: For input string: "false"
at scala.collection.immutable.StringLike$class$.parseBoolean(Unknown Source)
at scala.collection.TraversableLike$$anonfun$map$1.apply(Unknown Source)
at scala.collection.AbstractIterator.foreach(Unknown Source)
at scala.collection.TraversableLike$class$.map(Unknown Source)
at example.Main$.main(Unknown Source)
at <none>.main(Unknown Source)
at <none>.__libc_start_main(Unknown Source)
at <none>._start(Unknown Source)
at <none>.(Unknown Source)
java.lang.RuntimeException: Nonzero exit code: 1

Info:

- data: parsing the below line

203358,99103,false,3.6,1.0,,0,0.0,0.0,0.0,None,None,None,-1,0

- code in scala:

val objArr=line.split(",")

 objArr(2).toBoolean

**Comment by fsaab** [comment]
a side note: "false".toBoolean works on ubuntu installation but not on Mac. what i am missing?

**Comment by fsaab** [comment]
the same happened when i parse 0/1 as boolean:

java.lang.IllegalArgumentException: For input string: "0"
	at scala.collection.immutable.StringLike$class$.parseBoolean(Unknown Source)
	at org.ets.lca.Uncertainty.CalcExchange$.parse(Unknown Source)
	at org.ets.lca.Utilities.ResultsUtilsSingle$$anonfun$readExchnageCells$1.apply(Unknown Source)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(Unknown Source)
	at scala.collection.AbstractIterator.foreach(Unknown Source)
	at scala.collection.TraversableLike$class$.map(Unknown Source)
	at example.Main$.main(Unknown Source)
	at <none>.main(Unknown Source)
	at <none>.__libc_start_main(Unknown Source)
	at <none>._start(Unknown Source)
	at <none>.(Unknown Source)

**Comment by densh** [comment]
Given the code below:
```
object Test {
  def main(args: Array[String]): Unit = {
    val line = "203358,99103,false,3.6,1.0,,0,0.0,0.0,0.0,None,None,None,-1,0"
    val objArr = line.split(",")
    if (objArr(2).toBoolean) {
      println("FAIL")
    } else {
      println("SUCCESS")
    }
  }
}
```
I can reproduce the crash on `0.3.3` but the problem has been resolved in `0.3.6`. Please upgrade.
